### ISSUE TITLE
Password update with wildcard for certain username

### ISSUE THREAD

**Issue by aventrax** [roddhjav/pass-update#3]
Hello,

i have my domain password used to access various web interfaces.
The domain account has always the same password and it changes every 90 days.
My pass uses the structure as /web/www.example.com/username, and I would like to perform a "bulk update" of a certain username. I'm using git so I can't just update the first and copy the .gpg file overwriting the others, I'm expecting this to be possible:

pass update /web/*/username

Can this be implemented?

**Comment by roddhjav** [comment]
Yes it can, it's a good feature, thank for the proposal. I'll work on the implementation shortly

**Comment by roddhjav** [comment]
`pass update` can now update a full directory at once. However, bulk update as you wish pose an issue because when writing `pass update web/*/username`, the shell interprets `*` itself and passes the result to pass update.

**Comment by aventrax** [comment]
so should I quote the asterisk with a backslash or keep 'web/*/username' ?

**Comment by roddhjav** [comment]
Yes it could work however the auto-completion cannot work like this

**Comment by roddhjav** [comment]
Fixed with `pass-update 2.0`

**Comment by aventrax** [comment]
Finally the time to bulk changing my passwords has come.
I simply quoted the asterisk like this: `pass update -p "/web/*/mydomainuser"`

This worked, anyway it's pretty annoying process because pass asks..

>Changing password for /web/site1/mydomainuser
>oldpassword
>mydomainuser
>Are you ready to provide a new password? [y/N] **y**
>Enter the new password for /web/site1/mydomainuser: **..**
>Retype the new password for /web/site1/mydomainuser: **..**
>[master 0ae8c50] Update password for /web/site1/mydomainuser to store.
> 1 file changed, 0 insertions(+), 0 deletions(-)
> rewrite web/site1/mydomainuser.gpg (100%)

..and it repeat these lines for every founded password.
Plus, showing my old password is a non-sense because I do not need to see the old password and this costs me more sweat because of my Yubikey to be "touched" each time I had to decode something. This time could be saved if only it wouldn't show me my old passwords... 

I know that probably this is because of pass itself and not your extension, but do you know how to make this process easier for me?
Moreover, is there a way to provide the new password ONE TIME only, not having to repeat it each time?

**Comment by roddhjav** [comment]
You can speed up the process by doing:
```
pass update -m -f "/web/*/mydomainuser"
```

However, you will still need to provide the new passwords a lot of time. I don't plan to change this, because when using a password manager, we expect that all the passwords are different.

In your use case where this is the same user account, you should probably use password [inheritance](https://lists.zx2c4.com/pipermail/password-store/2017-October/003095.html).

**Comment by aventrax** [comment]
Right, I missed the option `-f` ;-) With it it's less painful :-)
Your decision to do not implement bulk-same-pwd-changing might be right, but when the user is changing a domain password it's doesn't look the best choice. Fortunately I have only a dozen domain password used by company's sites through browserpass, so it is manageable.
I'll look into inheritance the next time (3 months), because at the moment it seems just a "discussion" on the mailing list...

Many thanks.

**Comment by aventrax** [comment]
Hello, I'm here again because I found a better way to handle my domain password (the same on multiple web pages): The symlinks!

I created my user/pass as normal, then I manually create the *relative* symlinks on each folder I needed, than I add them to the GIT repo (manually). Links a managed by git, so on Linux I have no issues.

- pass insert /others/domain_user
- cd .password-store/web/blog.company.com/
- ln -s ../others/domain_user .
- git add domain_user
- git commit -m "test symlink"
- git push

Then, every time I need to change it, I only change /others/domain_user (not tried yet, but no reasons why it shouldn't work).

On windows I had the problem that GIT did not manage the symlinks, but only because I installed git scm without enabling "symlinks" during installation. Moreover, from Windows 10 creator update, the symlinks are enabled without special permissions, It only requires the "developer mode" enabled.

https://www.joshkel.com/2018/01/18/symlinks-in-windows/

So, here what I did on my Windows 10:

- Developer Mode enabled
- Git SCM with symlinks enabled
- Environment variable MSYS=winsymlinks:nativestrict
- git config --system core.symlinks true


Thats it!
